ibis/backends/flink/compiler.py

```diff
@@ -546,7 +546,7 @@ def visit_CountDistinct(self, op, *, arg, where):
         return self.f.count(sge.Distinct(expressions=[arg]))
 
     def visit_MapContains(self, op: ops.MapContains, *, arg, key):
-        return self.f.array_contains(self.f.map_keys(arg), key)
+        return arg[key].is_(NULL)
```

Comment:
You probably meant `arg[key].is_(sg.not_(NULL))`, but that's not correct either, for a different reason.

The code that was here before was intentionally written that way to handle the difference between a missing map key/value pair, and a map with an existing key whose value is `NULL`. Please revert this change. I will add a test in another PR to prevent other people from being tempted to alter this implementation :)

Reply:
I'll revert this and add a test.